File: cpp/104. 二叉树的最大深度.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    int ans=0;
    void traverse(TreeNode* root, int depth){
        if(root == nullptr){
            return;
        }
        ans=max(ans, depth);
        traverse(root->left, depth+1);
        traverse(root->right, depth+1);
    }
    int maxDepth(TreeNode* root) {
        traverse(root, 1);
        return ans;
    }
};

File: 刷题/golang/104. 二叉树的最大深度.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
 func maxDepth(root *TreeNode) int {
    if root == nil {
        return 0
    }
    left := maxDepth(root.Left)
    right := maxDepth(root.Right)
    res := max(left, right) + 1
    return res
}
func max(a, b int) int {
    if a > b {
        return a
    }
    return b
}